packages/driver/src/cy/net-stubbing/events/response.ts

```diff
@@ -104,9 +104,7 @@ export const onResponse: HandlerFn<CyHttpMessages.IncomingResponse> = async (Cyp
 
     finishResponseStage(res)
 
-    if (_.isObject(res.body)) {
-      res.body = JSON.stringify(res.body)
-    }
+    stringifyJsonObject(res)
```

Comment:
This is the main change. `_.isObject` returns true for many things including `ArrayBuffer`. The response body should only be stringified here if it was previously parsed, i.e. `hasJsonContentType` evaluated to true.